### ISSUE TITLE
Adding symbol whilst in 'text' mode

### ISSUE THREAD

**Issue by abrom** [daniel3735928559/guppy#127]
I've found that if you use the OSK to add a symbol whilst you're in 'text' mode, you end up causing the editor to go blank and some nice looking errors in the console:
```
Uncaught Error: KaTeX parse error: Can't use function '\sqrt' in text mode at position ....
```

What do you think about adding a method such as this in the `Engine`:
```
Engine.prototype.check_not_in_text = function(){
  // If we're not in text, return
  if (!Utils.is_text(this.current)) {
    return;
  }
  // If we are, then move to the next node
  this.sel_clear();
  // could use some recursion here just incase we have one text node following another??
  var nn = this.doc.xpath_node("following::e[1]", this.current);
  this.current = nn;
  this.caret = 0;
}
```

Then modify the OSK key click handler to call this before inserting symbols. I.e.
```
if(gn == "arithmetic" || gn == "qwerty" || gn == "QWERTY") guppy.engine.insert_string(n);
else {
  guppy.engine.check_not_in_text();
  guppy.engine.insert_symbol(n);
}
```

Not sure it's the most obvious way to do things. Better solution would probably be to 'split' the current text node and insert the new symbol in-between. 

Possibly might make sense to move the above logic into `insert_symbol` method, rather than having it separate?

**Comment by daniel3735928559** [comment]
Yeah--This is a bug. My thinking is that insert_symbol should be upgraded to handle it and, as you say, split the text node. It is a bit of a special case and might have to be applied to symbol nodes as well (although that might be more reasonable to just disallow).
